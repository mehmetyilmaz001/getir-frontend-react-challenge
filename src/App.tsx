import { useEffect } from "react";
import { useDispatch } from "react-redux";
import { useMediaQuery } from "react-responsive";
import CommonLayout from "./layouts/CommonLayout";
import ProductList from "./pages/ProductList/ProcuctList";
import { setIsBigScreen, setIsDesktopOrLaptop, setIsPortrait, setIsRetina, setIsTabletOrMobile } from "./redux/reducers/RootReducer";
import GlobalStyle from "./style/GlobalStyle";



function App() {

  const dispatch = useDispatch();

  const isTabletOrMobile = useMediaQuery({ maxWidth: 1224 })
  const isDesktopOrLaptop = useMediaQuery({ minWidth: 1224 })
  const isBigScreen = useMediaQuery({ minWidth: 1824 })
  const isPortrait = useMediaQuery({ orientation: 'portrait' })
  const isRetina = useMediaQuery({ minResolution: '2dppx' })

  useEffect(() => {
    dispatch(setIsDesktopOrLaptop(isDesktopOrLaptop));
    dispatch(setIsBigScreen(isBigScreen));
    dispatch(setIsTabletOrMobile(isTabletOrMobile));
    dispatch(setIsPortrait(isPortrait));
    dispatch(setIsRetina(isRetina));
  }, [
    isDesktopOrLaptop,
    isTabletOrMobile,
    isPortrait,
    isRetina,
    isBigScreen,
    dispatch
  ])

  return (
    <>
      <GlobalStyle />
      <CommonLayout>
        <ProductList />
      </CommonLayout>
    </>
  );
}

export default App;


