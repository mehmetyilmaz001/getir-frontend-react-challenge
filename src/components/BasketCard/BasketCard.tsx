import * as React from 'react';
import { useSelector } from 'react-redux';
import styled from 'styled-components';
import { Store } from '../../redux';
import theme from '../../style/Theme';
import Card from '../common/Card';
import BasketListItem from './components/BasketListItem/BasketListItem';
import BasketTotal from './components/BasketTotal/BasketTotal';


interface BasketCardProps {
}

const Container = styled.div`
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    gap: ${theme.spacing(4)};
`;


const TotalContainer = styled.div`
    width: 92px;
    height: 51px;
`
 

/**
 * Resuable component for displaying a basket which is connected to the redux store
 */
const BasketCard: React.FunctionComponent<BasketCardProps> = () => {
    
    const { items } = useSelector((state: Store) => state.basket);
    if(items.length === 0) {
        return <></>;
    }
    return ( 
        <Card style={{border: `solid 6px ${theme.palette.primary.main}`}}>
            <Container>
                {items.map((item, index) => <BasketListItem key={index} item={item} />)}

                <TotalContainer>
                    <BasketTotal showBasketIcon={false} theme='light'/>
               </TotalContainer>
            </Container>
        </Card>
     );
}
 
export default BasketCard;



 

 





 
