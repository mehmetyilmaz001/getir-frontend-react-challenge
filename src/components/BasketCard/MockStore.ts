import { Basket, BasketItem } from './../../types/Basket';
export const mockBasketItem: BasketItem = {
    added: 213213123,
    description: "",
    itemType: "",
    manufacturer: "",
    slug: "",
    tags: [],
    name: "test",
    price: 111,
    quantity: 1,
  };

export const mockBaskeCardStore: Basket = {
    items: [mockBasketItem],
    total: 2222
}