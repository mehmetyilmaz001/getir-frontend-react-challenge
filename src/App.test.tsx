import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import renderer from 'react-test-renderer';
import App from './App';  
import { mockBaskeCardStore, mockBasketItem } from "./components/BasketCard/MockStore";
const mockStore = configureStore([]);

test('App render test', () => {
  let store;
  let component: any;

    store = mockStore({
       root: {
          isTabletOrMobile: false,
       },
        basket: {
          ...mockBaskeCardStore
        },
        product: {
           data: {count: 0, items: [mockBasketItem]},
           pagination: {
             page:1
           }
        },
        lookup: {
          itemTypes: [],
        }
    });

    component = renderer.create(
        <Provider store={store}>
          <App />
        </Provider>
      );
 

  expect(component!.toJSON()).toMatchSnapshot();
});
