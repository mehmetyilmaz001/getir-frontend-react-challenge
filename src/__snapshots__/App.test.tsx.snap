// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render test 1`] = `
Array [
  <header
    className="sc-fotOHu gucLTB"
  >
    <div
      className="sc-fFeiMQ jBBNPc"
    >
      <img
        alt="Market"
        className="logo"
        src="SvgrURL"
      />
      <div
        className="sc-bkkeKt yCKlF"
        onClick={[Function]}
        role=""
      >
        <div
          className="sc-gsDKAQ kuDbVK"
        >
          <div />
          <span>
            ₺2.222,00
          </span>
        </div>
      </div>
    </div>
  </header>,
  <div
    className="sc-dlVxhl gNIAFh"
  >
    <div
      className="sc-llYSUQ bQnqyq"
    >
      <div
        className="sc-giYglK kZlJtp listing-column"
      >
        <h4
          className="sc-kLwhqv gnhWUy"
        >
          Products
        </h4>
        <div
          style={
            Object {
              "display": "flex",
              "gap": 8,
            }
          }
        />
        <div
          className="sc-bYoBSM kLLIaJ"
          role="grid"
        >
          <div
            className="sc-jJoQJp ieUIdf"
          >
            <img
              alt="test"
              className="sc-hiCibw gYKVcb"
              loading="lazy"
              src="https://picsum.photos/200/300?random=0.012096342240646818"
            />
            <span
              className="sc-gWXbKe kouSY"
            >
              ₺111,00
            </span>
            <h4
              className="sc-cCcXHH goTrCj"
            >
              test
            </h4>
            <button
              className="sc-eCImPb kWJqWx"
              onClick={[Function]}
            >
              Add
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <footer
    className="sc-bdvvtL ihGqeD"
  >
    ©
    2021
     Market 
    <span
      style={
        Object {
          "fontSize": "24px",
          "fontWeight": "bold",
          "margin": "0 16px",
        }
      }
    >
      ·
    </span>
     Privacy Policy
  </footer>,
]
`;
